index.js

```diff
@@ -33,14 +31,9 @@ MessageBuffer.prototype.drain = function(cb) {
   var oldBuffer = this.buffer;
   this.buffer = [];
 
-  if (this.messagesAdded > oldBuffer.length) {
-    var startIdx = this.messagesAdded % oldBuffer.length;
-    for (var i=startIdx; i<(oldBuffer.length + startIdx); i++) {
-      cb(oldBuffer[i % this.messageMax]);
-    }
-  } else {
-    oldBuffer.forEach(cb);
-  }
+  oldBuffer.forEach(cb);
+
+  this.messagesAdded = 0;
```

Comment:
After looking at this some more, this may have been the only required change. I would say the whole index thing is a little simpler now, but that's pretty subjective.